packages/mars/src/syntax/createProxy.ts

```diff
@@ -0,0 +1,106 @@
+import { Contract, contract, makeContractInstance, Options } from './contract'
+import { Artifact, ArtifactNoParams, Params } from './artifact'
+import { Address, ArtifactSymbol, Constructor, Methods, Name } from '../symbols'
+import { BooleanLike } from '../values'
+import { context } from '../context'
+import { constants } from 'ethers'
+
+export type Proxy = (<T extends Artifact, U extends keyof T[typeof Methods]>(
```

Comment:
You can also use
```typescript
interface Callable {
   (foo: string) => void;
   (bar: number) => void;
}
```
to define overloaded function types in an IMHO bit cleaner way

Reply:
thanks, didn't know that